classes/userGroup/Collector.php

```diff
@@ -0,0 +1,288 @@
+        $this->userIds = $userIds;
+        return $this;
+    }
+
+    /**
+     * Include orderBy columns to the collector query
+     */
+    public function orderBy(?string $orderBy): self
+    {
+        $this->orderBy = $orderBy;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(?int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(?int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = DB::table('user_groups as a');
+
+        if (isset($this->ids)) {
+            $q->whereIn('a.user_group_id', $this->ids);
+        }
+
+        if (isset($this->userIds)) {
+            $q->join('user_user_groups as uug', 'a.user_group_id', '=', 'uug.user_group_id');
+            $q->whereIn('uug.user_id', $this->userIds);
+        }
+
+        if (isset($this->stageIds)) {
+            $q->join('user_group_stage as ugs', function($join) {
+                $join->on('a.user_group_id', '=', 'ugs.user_group_id');
+                $join->on('a.context_id','=', 'ugs.context_id');
+            })->whereIn('ugs.stage_id', $this->stageIds);
+        }
+
+        if (isset($this->contextIds)) {
+            $q->whereIn('a.context_id', $this->contextIds);
+        }
+
+        if (isset($this->roleIds)) {
+            $q->whereIn('a.role_id', $this->roleIds);
+        }
+
+        $q->when($this->isRecommendOnly !== null, function (Builder $q) {
+            $q->whereIn('a.user_group_id', function (Builder $q) {
+                $q->select('user_group_id')
+                    ->from($this->dao->settingsTable)
+                    ->where('setting_name', '=', 'recommendOnly')
+                    ->where('setting_value', $this->isRecommendOnly);
+            });
+        });
+
+        if ($this->isDefault) {
+            $q->where('a.is_default', $this->isDefault);
+        }
+
+        if ($this->permitSelfRegistration) {
+            $q->where('a.permit_self_registration', $this->permitSelfRegistration);
+        }
+
+        if ($this->permitMetadataEdit) {
+            $q->where('a.permit_metadata_edit', $this->permitMetadataEdit);
+        }
+
+        if ($this->showTitle) {
+            $q->where('a.show_title', $this->showTitle);
+        }
+
+        if (isset($this->count)) {
+            $q->limit($this->count);
+        }
+
+        if (isset($this->offset)) {
+            $q->offset($this->offset);
+        }
+
+        switch ($this->orderBy) {
+             case self::ORDERBY_ROLE_ID:
+                $q->orderBy('a.role_id', 'asc');
+                break;
+            case self::ORDERBY_ID:
+            default:
```

Comment:
I think I would prefer...
- Initialize `$orderBy` to `null` at the top of the class (for no sorting)
- Have `default:` apply no sorting at al

This would not apply sorting where it's not needed (better for performance) and allow for subclasses or plugins to extend it with additional options if needed.

Reply:
- `$orderBy` was already initialized to `null` (am I missing something?)
- Change the Switch clause removing the `default:` case.

Reply:
> am I missing something?

Nope, it looks good!